test/steps/kogitoapp.go

```diff
@@ -185,31 +157,145 @@ func getKogitoAppExamplesStub(namespace, contextDir string, native, persistence,
+
+		case RuntimeRequest:
+			kogitoApp.Spec.AddResourceRequest(getSecondColumn(row), getThirdColumn(row))
+
+		case RuntimeLimit:
+			kogitoApp.Spec.AddResourceLimit(getSecondColumn(row), getThirdColumn(row))
+
+		default:
+			return fmt.Errorf("Unrecognized configuration option: %s", firstColumn)
+		}
 	}
 
 	if len(profiles) > 0 {
-		kogitoApp.Spec.Build.AddEnvironmentVariable(mavenArgsAppendEnvVar, "-P"+strings.Join(profiles, ","))
+		kogitoApp.Spec.Build.AddEnvironmentVariable(MavenArgsAppendEnvVar, "-P"+strings.Join(profiles, ","))
 	}
 
-	if native {
-		kogitoApp.Spec.Build.Native = native
-		// Make sure that enough memory is allocated for builder pod in case of native build
-		kogitoApp.Spec.Build.Resources = corev1.ResourceRequirements{
-			Requests: corev1.ResourceList{
-				v1.ResourceName("memory"): resource.MustParse("4Gi"),
-			},
+	addDefaultJavaOptionsIfNotProvided(kogitoApp)
+
+	return nil
+}
+
+func parseConfigRow(row *gherkin.TableRow, kogitoApp *v1alpha1.KogitoApp, profilesPtr *[]string) {
+	secondColumn := getSecondColumn(row)
+
+	switch secondColumn {
+	case Native:
+		native := framework.MustParseEnabledDisabled(getThirdColumn(row))
+		if native {
+			kogitoApp.Spec.Build.Native = native
+			// Make sure that enough memory is allocated for builder pod in case of native build
+			kogitoApp.Spec.Build.AddResourceRequest("memory", "4Gi")
+		}
+
+	case Persistence:
+		persistence := framework.MustParseEnabledDisabled(getThirdColumn(row))
+		if persistence {
+			*profilesPtr = append(*profilesPtr, "persistence")
+			kogitoApp.Spec.EnablePersistence = true
+		}
+
+	case Events:
+		events := framework.MustParseEnabledDisabled(getThirdColumn(row))
+		if events {
+			*profilesPtr = append(*profilesPtr, "events")
+			kogitoApp.Spec.EnableEvents = true
+			kogitoApp.Spec.KogitoServiceSpec.AddEnvironmentVariable("MP_MESSAGING_OUTGOING_KOGITO_PROCESSINSTANCES_EVENTS_BOOTSTRAP_SERVERS", "")
+			kogitoApp.Spec.KogitoServiceSpec.AddEnvironmentVariable("MP_MESSAGING_OUTGOING_KOGITO_USERTASKINSTANCES_EVENTS_BOOTSTRAP_SERVERS", "")
 		}
 	}
+}
 
-	return kogitoApp
+func addDefaultJavaOptionsIfNotProvided(kogitoApp *v1alpha1.KogitoApp) {
+	javaOptionsProvided := false
+	for _, env := range kogitoApp.Spec.Envs {
+		if env.Name == JavaOptionsEnvVar {
+			javaOptionsProvided = true
+		}
+	}
+
+	if !javaOptionsProvided {
+		kogitoApp.Spec.AddEnvironmentVariable(JavaOptionsEnvVar, "-Xmx2G")
+	}
+}
+
+func getFirstColumn(row *gherkin.TableRow) string {
+	return row.Cells[0].Value
+}
+
+func getSecondColumn(row *gherkin.TableRow) string {
+	return row.Cells[1].Value
+}
+
+func getThirdColumn(row *gherkin.TableRow) string {
+	return row.Cells[2].Value
+}
+
+// parseResourceRequirementsTable is useful for steps that check resource requirements, table is a subset of KogitoApp
+// configuration table
+func parseResourceRequirementsTable(table *gherkin.DataTable) (build, runtime *v1.ResourceRequirements, err error) {
```

Comment:
I think we're duplicating how we parse the resources in three places:
- In the Build type (for build resources)
- In the Spec type (for runtime resources)
- In this method (to assert resources in tests)

What about only using this method to set the resources either in build and spec types, and also in the asserts?

Reply:
Yes, I am aware of that. The reason why I split it was to not parse the table multiple times. Wdyt? If I reused this part I would have to parse the table again after already reading it :) In my opinion, if we reuse this, there will be less duplication but the final code will be less readable than now.

Reply:
I'm leaving this up to you, so ignore this if you disagree. From my point of view, if we only use this method in this file, I would prefer only having it in this file.

Reply:
So you are suggesting that this shouldn't go to the `utils` file with the configureKogitoAppFromTable function?

Reply:
I'm suggesting to keep this method only and reuse it.

Reply:
I know, but this confused me: 
> From my point of view, if we only use this method in this file, I would prefer only having it in this file.

Not sure what you meant by that.

Reply:
I meant only using this method.